src/lib/data/github.ts

```diff
@@ -32,25 +33,130 @@ export class GitHubCollector {
+        repoData,
+        repoContents,
+        readmeContent,
+      };
     } catch (error) {
       throw new Error(`GitHub fetch failed: ${error}`);
     }
   }
 
-  getStarCount(): number {
-    return this.starCount || 0;
+  private async fetchAllRepoContents(owner: string, repo: string): Promise<Record<string, boolean>> {
+    const results: Record<string, boolean> = {};
+    const visitedPaths = new Set<string>();
+
+    try {
+      // Start recursive fetch from root
+      await this.fetchDirectoryContentsRecursive(owner, repo, '', results, visitedPaths);
+    } catch (error) {
+      console.warn(`Failed to fetch repository contents: ${error}`);
+    }
+
+    return results;
   }
 
-  getData(): GitHubData {
-    return {
-      stars: this.getStarCount(),
-    };
+  private async fetchDirectoryContentsRecursive(
+    owner: string,
+    repo: string,
+    path: string,
+    results: Record<string, boolean>,
+    visitedPaths: Set<string>,
+    maxDepth: number = 3,
+    currentDepth: number = 0,
+  ): Promise<void> {
+    // Prevent infinite recursion and limit depth, also should not need
+    // to recurse more than 3 layers for the  documentation?
+    if (currentDepth >= maxDepth || visitedPaths.has(path)) {
+      return;
+    }
+
+    visitedPaths.add(path);
+
+    try {
+      const response = await fetch(`https://api.github.com/repos/${owner}/${repo}/contents/${path}`);
+      if (!response.ok) return;
+
+      const data = await response.json() as any;
+      if (!Array.isArray(data)) return;
+
+      // Process all items in parallel
+      const directoryPromises: Promise<void>[] = [];
+
+      for (const item of data) {
+        const itemPath = path ? `${path}/${item.name}` : item.name;
+        results[itemPath] = true;
+
+        // Also add just the filename for easier matching
+        results[item.name] = true;
+
+        // If it's a directory, recursively fetch its contents
+        if (item.type === 'dir') {
+          directoryPromises.push(
+            this.fetchDirectoryContentsRecursive(
+              owner,
+              repo,
+              itemPath,
+              results,
+              visitedPaths,
+              maxDepth,
+              currentDepth + 1,
+            ),
+          );
+        }
+      }
+
+      // Wait for all subdirectory fetches to complete
+      await Promise.all(directoryPromises);
+    } catch (error) {
+      // Silently fail for individual directory fetches
+    }
+  }
+
+  private async fetchReadmeContent(owner: string, repo: string, repoContents: Record<string, boolean>): Promise<string | null> {
+    // Find README file from the repository contents
+    const readmeFile = Object.keys(repoContents).find(path =>
+      path.toLowerCase().includes('readme') && !path.includes('/'),
+    );
+
+    if (!readmeFile) {
+      return null;
+    }
+
+    try {
+      const response = await fetch(`https://api.github.com/repos/${owner}/${repo}/contents/${readmeFile}`);
```

Comment:
you're writing this everywhere, you should have a helper class for fetching from github. to start you off, it might look something like this.

```ts
class GitHubRepo {
  private url;
  constructor(owner: string, repo: string) {
    this.url = `https://api.github.com/repos/${owner}/${repo}`;
  }

  async metadata() {
    return fetch(this.url);
  }

  async file(file: string) {
    return fetch(`${this.url}/contents/${file}`);
  }
}
```

this should be in a separate refactor PR, and probably do this first because all your other PRs depend on this. you also need to consider how to gracefully fail if `fetch` returns error.